python/lsst/ctrl/mpexec/cmdLineFwk.py

```diff
@@ -351,6 +353,30 @@ def makeButlerAndCollections(cls, args: SimpleNamespace) -> Tuple[Butler, Sequen
         _LOG.debug("Preparing registry to read from %s and expect future writes to '%s'.", inputs, run)
         return butler, inputs, run
 
+    @staticmethod
+    def defineDatastoreCache() -> None:
+        """Define where datastore cache directories should be found.
+
+        Notes
+        -----
+        All the jobs should share a datastore cache if applicable. This
+        method determines a name and sets an environment variable that
+        will be picked up by the caching class. A value will not be set
+        if one is already defined.
+        """
+        if os.environ.get("DAF_BUTLER_CACHE_DIRECTORY_IF_UNSET"):
```

Comment:
Would it be better to hide this sort of details (envvar name) in `DatastoreCacheManager` class? Or maybe even the whole method?

Reply:
I think I do still need to set an environment variable somewhere since that's the only thing we guarantee will end up in a spawned subprocess without having to put code in the executeJob method. The DatastoreCacheManager class is a bit of an internal detail but I'll see about adding a `DatastoreCacheManager.set_fallback_cache_dir()` method that still uses the environment.